addons/project/models/project.py

```diff
@@ -710,6 +701,29 @@ def _change_privacy_visibility(self):
             project.message_unsubscribe(partner_ids=portal_users.partner_id.ids)
             project.mapped('tasks')._change_project_privacy_visibility()
 
+    # ---------------------------------------------------
+    # Project sharing
+    # ---------------------------------------------------
+    def _check_project_sharing_access(self):
+        self.ensure_one()
+        if self.env.user.has_group('base.group_portal'):
+            return self.env.user.partner_id in self.collaborator_ids.partner_id
```

Comment:
Shouldn't the `privacy_visibility` be involved here?

Reply:
Yes, it should, thanks